### ISSUE TITLE
Cannot connect to database

### ISSUE THREAD

**Issue by maylukas** [playframework/play-slick#333]
I tried to run the example app you provided in your samples but unfortunately the plugin fails to connect to the database although it doesn't provide any logs about errors or similar.

**My DAO file (Then called from the controller)**:
```scala
class CatDAO @Inject()(@NamedDatabase("default") val dbConfigProvider: DatabaseConfigProvider) extends HasDatabaseConfig[JdbcProfile] {
  protected val dbConfig = DatabaseConfigProvider.get[JdbcProfile](Play.current)
  import driver.api._
 ...
def insert(cat: Cat): Future[Unit] = {

    db.run(Cats += cat).map { _ => () }
  }
```

**My configuration is as follows**:
```
slick.dbs.default.driver="slick.driver.MySQLDriver$"
slick.dbs.default.db.driver="com.mysql.jdbc.Driver"
slick.dbs.default.db.url="jdbc:mysql://127.0.0.1:3306/mydb"
slick.dbs.default.db.username="myusername"
slick.dbs.default.db.password="mypassword"
slick.dbs.default.db.connectionTestQuery="select version();"
```

When I enabled logging for "com.zaxxer.hikari" i noticed that the following error occured roughly every second:
```
java.sql.SQLException: Access denied for user ''@'localhost' (using password: YES)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:959) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3870) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3806) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:871) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.MysqlIO.proceedHandshakeWithPluggableAuthentication(MysqlIO.java:1686) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.MysqlIO.doHandshake(MysqlIO.java:1207) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2254) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2285) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2084) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:795) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:44) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at sun.reflect.GeneratedConstructorAccessor26.newInstance(Unknown Source) ~[na:na]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.8.0_25]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:408) ~[na:1.8.0_25]
        at com.mysql.jdbc.Util.handleNewInstance(Util.java:404) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:400) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:327) ~[mysql-connector-java-5.1.37.jar:5.1.37]
        at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:92) ~[HikariCP-java6-2.3.7.jar:na]
        at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:98) ~[HikariCP-java6-2.3.7.jar:na]
        at com.zaxxer.hikari.pool.BaseHikariPool.addConnection(BaseHikariPool.java:438) ~[HikariCP-java6-2.3.7.jar:na]
        at com.zaxxer.hikari.pool.BaseHikariPool$1.run(BaseHikariPool.java:413) [HikariCP-java6-2.3.7.jar:na]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_25]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_25]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_25]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_25]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_25]
```

I then tried to additionally configure the database like so:
```
db.default.driver=com.mysql.jdbc.Driver
db.default.url="jdbc:mysql://127.0.0.1:3306/mydb"
db.default.user=myusername
db.default.password="mypassword"
```

Unfortunately this did not have any effect too.

I am using Play 2.4.3 and my dependencies are:
```scala
libraryDependencies ++= Seq(
  cache,
  ws,
  specs2 % Test,
  "com.typesafe.play" %% "play-slick" % "1.1.1",
  "mysql" % "mysql-connector-java" % "5.1.37",
  "com.typesafe.slick" %% "slick-codegen" % "3.1.0" % "compile"
)
```

I don't know if this is of interest but when turning on logging for the package slick I receive the following messages:

```
[debug] - play.api.db.slick.DefaultSlickApi - Created Slick database config for key default.
[info] - play.api.Play - Application started (Dev)
[success] Compiled in 54ms
[debug] - slick.compiler.QueryCompiler - Source:
| TableExpansion
|   table s2: Table cat
|   columns: TypeMapping
|     0: ProductNode
|       1: Path s2.name : String'
|       2: Path s2.color : String'

[debug] - slick.compiler.AssignUniqueSymbols - Detected features: UsedFeatures(false,true,false,false)
[debug] - slick.compiler.QueryCompiler - After phase assignUniqueSymbols:
| TableExpansion
|   table s3: Table cat
|   columns: TypeMapping
|     0: ProductNode
|       1: Path s3.name : String'
|       2: Path s3.color : String'

[debug] - slick.compiler.QueryCompiler - After phase inferTypes: (no change)
[debug] - slick.compiler.QueryCompiler - After phase insertCompiler:
| ResultSetMapping : Vector[(String', String')]
|   from s5: Insert allFields=[name, color] : (String', String')
|     table s6: Table cat : Vector[@t4<UnassignedType>]
|     linear: ProductNode : (String', String')
|       1: Path s6.name : String'
|       2: Path s6.color : String'
|   map: TypeMapping : Mapped[(String', String')]
|     0: ProductNode : (String', String')
|       1: InsertColumn name : String'
|         0: Path s5._1 : String'
|       2: InsertColumn color : String'
|         0: Path s5._2 : String'

[debug] - slick.compiler.CodeGen - Compiling server-side and mapping with server-side:
| Insert allFields=[name, color] : (String', String')
|   table s6: Table cat : Vector[@t4<UnassignedType>]
|   linear: ProductNode : (String', String')
|     1: Path s6.name : String'
|     2: Path s6.color : String'

[debug] - slick.relational.ResultConverterCompiler - Compiled ResultConverter
| TypeMappingResultConverter
|   child: ProductResultConverter
|     1: SpecializedJdbcResultConverter$$anon$1 idx=1, name=name : String'
|     2: SpecializedJdbcResultConverter$$anon$1 idx=2, name=color : String'

[debug] - slick.compiler.CodeGen - Compiled server-side to:
| CompiledStatement "insert into `cat` (`name`,`color`)  values (?,?)" : (String', String')

[debug] - slick.compiler.QueryCompiler - After phase codeGen:
| ResultSetMapping : Vector[(String', String')]
|   from s5: CompiledStatement "insert into `cat` (`name`,`color`)  values (?,?)" : (String', String')
|   map: CompiledMapping : Mapped[(String', String')]
|     converter: TypeMappingResultConverter
|       child: ProductResultConverter
|         1: SpecializedJdbcResultConverter$$anon$1 idx=1, name=name : String'
|         2: SpecializedJdbcResultConverter$$anon$1 idx=2, name=color : String'

[debug] - slick.compiler.QueryCompilerBenchmark - ------------------- Phase: Time ---------
[debug] - slick.compiler.QueryCompilerBenchmark -       assignUniqueSymbols:    0.627752 ms
[debug] - slick.compiler.QueryCompilerBenchmark -                inferTypes:    0.064425 ms
[debug] - slick.compiler.QueryCompilerBenchmark -            insertCompiler:    1.429918 ms
[debug] - slick.compiler.QueryCompilerBenchmark -                   codeGen:    1.443648 ms
[debug] - slick.compiler.QueryCompilerBenchmark -                     TOTAL:    3.565743 ms
[debug] - slick.backend.DatabaseComponent.action - #1: SingleInsertAction [insert into `cat` (`name`,`color`)  values (?,?)]
[debug] - slick.jdbc.JdbcBackend.statement - Preparing statement: insert into `cat` (`name`,`color`)  values (?,?)
```

Is there anything that is missed in my configuration or implementation? I hope I am not missing something completely obvious :see_no_evil: 
I would be happy if anyone could help me with this.

**Comment by dotta** [comment]
Hi @maylukas! The issue tracker is not the right place for asking usage questions. I'm closing this ticket, please use the [playframework user mailing list](https://groups.google.com/forum/#!forum/play-framework) instead.

There is currently a problem with running the sample projects, see #332. However, it is not related to what you have described in the above ticket. Here is how you can run the `dao-sample` project:

```
$ sbt
[info] Loading global plugins from /Users/mirco/.sbt/0.13/plugins
[info] Updating {file:/Users/mirco/.sbt/0.13/plugins/}global-plugins...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Loading project definition from /Users/mirco/Projects/oos/playframework/play-slick/project
[info] Updating {file:/Users/mirco/Projects/oos/playframework/play-slick/project/}play-slick-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/mirco/.ivy2/cache/org.slf4j/slf4j-simple/jars/slf4j-simple-1.7.12.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/mirco/.ivy2/cache/org.slf4j/slf4j-nop/jars/slf4j-nop-1.7.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
[info] Set current project to play-slick-root (in build file:/Users/mirco/Projects/oos/playframework/play-slick/)
> projects
[info] In file:/Users/mirco/Projects/oos/playframework/play-slick/
[info] 	   basic-sample
[info] 	   computer-database-sample
[info] 	   dao-sample
[info] 	   di-sample
[info] 	   docs
[info] 	   iteratee-sample
[info] 	   json-sample
[info] 	   play-slick
[info] 	   play-slick-evolutions
[info] 	 * play-slick-root
[info] 	   samples
> project dao-sample
[info] Set current project to dao-sample (in build file:/Users/mirco/Projects/oos/playframework/play-slick/)
[dao-sample] $ ++ 2.11.7
[info] Setting version to 2.11.7
[info] Reapplying settings...
[info] Set current project to dao-sample (in build file:/Users/mirco/Projects/oos/playframework/play-slick/)
[dao-sample] $ run
--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.NettyServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000

```

Hence, the sample project in master works just fine.

From the error, it looks like the specified user doesn't have access to the db (maybe you need to create the user in the db first).

**Comment by maylukas** [comment]
@dotta: I think I was not precise concerning my description. I am sorry about that.
Whenever I try to insert a row into the database the query gets executed and no exception is thrown but it has no effect on the database.
The user I configured does exist in the database and the password is also correct.
I only found the exception when I enabled logging for the package "com.zaxxer.hikari" which apperently does not pick up the database configuration I provided for slick.
I used the example application because I wanted to omit any implementation issues that I might have made.
